flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapFoldingState.java

```diff
@@ -49,17 +49,18 @@
 	/**
 	 * Creates a new key/value state for the given hash map of key/value pairs.
 	 *
-	 * @param stateDesc The state identifier for the state. This contains name
-	 *                           and can create a default state value.
+	 * @param valueSerializer The serializer for the state.
 	 * @param stateTable The state tab;e to use in this kev/value state. May contain initial state.
 	 */
 	public HeapFoldingState(
-			FoldingStateDescriptor<T, ACC> stateDesc,
 			StateTable<K, N, ACC> stateTable,
 			TypeSerializer<K> keySerializer,
-			TypeSerializer<N> namespaceSerializer) {
-		super(stateDesc, stateTable, keySerializer, namespaceSerializer);
-		this.foldTransformation = new FoldTransformation<>(stateDesc);
+			TypeSerializer<ACC> valueSerializer,
+			TypeSerializer<N> namespaceSerializer,
+			ACC defaultValue,
```

Comment:
Comments require update.

Reply:
I think you need to double check this on every state class, they look all not updated.

Reply:
Javadocs for all state classes are now updated.